### PR TITLE
Beats docs - Contributing improvements

### PR DESCRIPTION
**Pull request by odacremolbap**
A very easy rookie contribution:
- Fix a link to PR review. Solves #11069 
- Change recommended go version for development
- Change clone example to use SSH

### CODE REVIEW

**Comment by ruflin** [comment]
The docs CI failure on Jenkins is strange as it passes on the others. I will retrigger the build.

**Comment by ruflin** [comment]
jenkins, test this

File: docs/devguide/contributing.asciidoc [hunk 1]
```diff
@@ -56,7 +55,7 @@ simply set it to the `go` directory in your home (`export GOPATH=$HOME/go`).
 --------------------------------------------------------------------------------
 mkdir -p ${GOPATH}/src/github.com/elastic
 cd ${GOPATH}/src/github.com/elastic
-git clone https://github.com/elastic/beats.git
+git clone git@github.com:elastic/beats.git
```
Comment (ruflin) [inline docs/devguide/contributing.asciidoc]: I would keep `https` here as it's a public repo and I personally would recommend to pull it over https and only the clone over git by default. Like this you never push accidentally to the elastic repo but always work with PR's.

Obviously if someone needs to create branches / tags this is different but very few contributors need to do this.

Comment (odacremolbap) [reply docs/devguide/contributing.asciidoc]: sounds good, reverting that change
Thanks!

**Pull request merged**